src/Database/LSMTree/Internal/Lookup.hs

```diff
@@ -230,7 +231,14 @@ data ByteCountDiscrepancy = ByteCountDiscrepancy {
 -- PRECONDITION: the vectors of bloom filters, indexes and file handles
 -- should pointwise match with the vectors of runs.
 lookupsIO ::
-     forall m h. (PrimMonad m, MonadThrow m, MonadST m)
+     forall m h.
+     (
+#if MIN_VERSION_io_classes(1,5,0)
+-- Since io-classes-1.5.0.0 the MonadST type-class constraint provides MonadPrim m => MonadST m
+#else
+     PrimMonad m,
+#endif
```

Comment:
The CPP is redundant, since we're always using `io-classes-1.5.0.0`

Reply:
> You can probably just remove the local one. It was only there because writeTMVar was missing from io-classes

I removed the local definition of `writeTMVar` in favor of the exported definition from `io-classes`.

> The CPP is redundant, since we're always using io-classes-1.5.0.0

An excellent observation. It is best to remove CPP usage wherever possible. I have indeed removed the superfluous usage in the `Database.LSMTree.Internal.Lookup` module.